### DISCUSSION TITLE
Shouldn't the enums and interfaces be exported?

### DISCUSSION THREAD

**Discussion by Vithanco** [hpcc-systems/hpcc-js-wasm#91] [Ideas]
My code cannot fully use these declarations as they are not exported:
```
declare type Format = "svg" | "dot" | "json" | "dot_json" | "xdot_json" | "plain" | "plain-ext";
declare type Engine = "circo" | "dot" | "fdp" | "sfdp" | "neato" | "osage" | "patchwork" | "twopi";
interface Image {
    path: string;
    width: string;
    height: string;
}
interface File {
    path: string;
    data: string;
}
interface Ext {
    images?: Image[];
    files?: File[];
    wasmFolder?: string;
    wasmBinary?: Uint8Array;
    yInvert?: boolean;
    nop?: number;
}
```

Is that an oversight or do I not understand how to use the library? Background: I am creating a wrapper for your Wasm that will fit my needs (based on JSON output) and I realised that I cannot create a wrapper with the same types as these types aren't exported.

**Comment by GordonSmith** [comment]
@Vithanco those types are exported:
1. In package.json:  `  "types": "types/index.d.ts",`
2. You can see the types here:  https://cdn.jsdelivr.net/npm/@hpcc-js/wasm@1.12.8/types/graphviz.d.ts

**Comment by Vithanco** [comment]
hmmm. Maybe I am too new to JavaScript. But I cannot see these types being
exported. They are there but lack the export keyword.

On Sat, Feb 5, 2022 at 9:40 AM Gordon Smith ***@***.***>
wrote:

> @Vithanco <https://github.com/Vithanco> those types are exported:
>
>    1. In package.json: "types": "types/index.d.ts",
>    2. You can see the types here:
>    ***@***.******@***.***/types/graphviz.d.ts
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hpcc-systems/hpcc-js-wasm/discussions/91#discussioncomment-2115851>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABFFOMFREMIYPSR2RN3HSL3UZTO6NANCNFSM5NTSOPIQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Klaus Kneupner <http://www.kneupner.de>
"Denn von allen Gedanken, schätz' ich doch am meisten, die interessanten"
Die Sterne

**Reply by GordonSmith** [reply]
You are correct they are "visible" but not exported. 

So you are extending the graphviz class and potentially want to extend those interfaces / types as well?

**Comment by GordonSmith** [comment]
See:  https://github.com/hpcc-systems/hpcc-js-wasm/issues/92

**Reply by Vithanco** [reply]
this was fast. Much appreciated! Thank you.
